components/utilities/floatingNav.js

```diff
@@ -112,11 +149,43 @@ const Headings = ({ headings, activeId }) => {
 };
 
 const Heading = ({ heading, index, activeId }) => {
-  const active = heading.target === activeId ? "active" : "";
+  const active =
+    heading.target === activeId
+      ? `${floatingNavStyles.activeItem} dark:text-white`
+      : "";
 
   return (
-    <li className={`level-${heading.level} ${active}`} key={`toc-${index}`}>
-      <a href={heading.target}>{heading.label}</a>
+    <li
+      className={`
+        m-0
+        text-xs
+        pt-2
+        ${
+          heading.level === "H1" || heading.level === "H2"
+            ? "pl-6"
+            : heading.level === "H3"
+            ? "pl-8"
+            : heading.level === "H4"
+            ? "pl-10"
+            : heading.level === "H5"
+            ? "pl-12"
+            : "pl-14"
```

Comment:
Move the above into a class.


For the heading picker, you could go different routes:

heading.level == "H1" ? styles.headingH1 : etc

styles[heading.level]

use "heading1" as a style
:global(heading1)

Reply:
Fixed on c8573ad